opm/simulators/flow/FlowProblemBlackoil.hpp

```diff
@@ -0,0 +1,1270 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+// vi: set et ts=4 sw=4 sts=4:
+/*
+  Copyright 2023 INRIA
+  Copyright 2024 SINTEF Digital
+  
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+
+  Consult the COPYING file in the top-level source directory of this
+  module for the precise wording of the license and the list of
+  copyright holders.
+*/
+/*!
+ * \file
+ *
+ * \copydoc Opm::FlowProblemBlackoil
+ */
+#ifndef OPM_FLOW_PROBLEM_BLACK_HPP
+#define OPM_FLOW_PROBLEM_BLACK_HPP
+
+#include <opm/material/fluidsystems/BlackOilFluidSystem.hpp>
+#include <opm/material/fluidsystems/blackoilpvt/DryGasPvt.hpp>
+#include <opm/material/fluidsystems/blackoilpvt/WetGasPvt.hpp>
+#include <opm/material/fluidsystems/blackoilpvt/LiveOilPvt.hpp>
+#include <opm/material/fluidsystems/blackoilpvt/DeadOilPvt.hpp>
+#include <opm/material/fluidsystems/blackoilpvt/ConstantCompressibilityOilPvt.hpp>
+#include <opm/material/fluidsystems/blackoilpvt/ConstantCompressibilityWaterPvt.hpp>
+
+#include <opm/models/blackoil/blackoillocalresidualtpfa.hh>
+
+//#include <opm/output/eclipse/EclipseIO.hpp>
+
+#include <opm/simulators/flow/FlowProblem.hpp>
+
+#include <opm/simulators/flow/MixingRateControls.hpp>
+
+#include <opm/simulators/flow/ActionHandler.hpp>
+#if HAVE_DAMARIS
+#include <opm/simulators/flow/DamarisWriter.hpp>
+#endif
+
+#include <algorithm>
+#include <functional>
+#include <set>
+#include <string>
+#include <vector>
+
+namespace Opm {
+
+/*!
+ * \ingroup BlackOilSimulator
+ *
+ * \brief This problem simulates an input file given in the data format used by the
+ *        commercial ECLiPSE simulator.
+ */
+template <class TypeTag>
+class FlowProblemBlackoil : public FlowProblem<TypeTag>
+{
+    // TODO: the naming of the Types will be adjusted
+    using FlowProblemType = FlowProblem<TypeTag>;
```

Comment:
Yeah, this is easy to do. I can rename if you think it is a good idea.

The naming is a little difficult here, we have the `FlowBaseProblem` in the TypeTag. And also, we have a couple class type and TypeTag share the same name, I found it is difficult. 

Maybe some systematic change can be done.

Reply:
Yeah. I misunderstood your comment. 

I can rename `FlowProblemType` to `FlowProblemBase`.

Reply:
Hei again. still because the existence of `FlowBaseProblem`, I decided to keep `FlowProblemType`.  It is not that important, but prefer this way after thinking back and forth.  let me know if you have strong opinion here.